apache/incubator-zeppelin
# Resource pools api

### What is this PR for?
This is a step towards creating a REST endpoint from resource pools  https://issues.apache.org/jira/browse/ZEPPELIN-699

It allows resources to be downloaded from results using a REST API

### What type of PR is it?
Improvement

### Todos
* [X] - Add ability to download resources via an API endpoint

### What is the Jira issue?
Finishes:  https://issues.apache.org/jira/browse/ZEPPELIN-699

Note, that this pull request is based upon these:  
https://github.com/apache/incubator-zeppelin/pull/782
https://github.com/apache/incubator-zeppelin/pull/816

### How should this be tested?
1.  Edit an interpreter
2.  Add a new property to the interpreter - the "ResourcePoolClass" - and assign it to "org.apache.zeppelin.resource.VFSResourcePool" 
3.  Run the interpreter

You should see new resource files created in notebook/zeppelin_resources

Then go to 
`/api/notebook/results`
to see the rest endpoint that lists all of the results.

View it with
`/api/notebook/[noteid]/paragraph/[paragraph id]/results`

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation?  No.
